File: 232.cpp
struct MyQueue {
    vector<int> head, tail;
    MyQueue(): head(), tail() {}
    
    inline void push(int x) {
        this->tail.push_back(x);
    }
    
    int pop() {
        this->peek();
        int val = this->head.back();
        this->head.pop_back();
        return val;
    }
    
    int peek() {
        if (this->head.empty()){
            while (!this->tail.empty()){
                this->head.push_back(this->tail.back());
                this->tail.pop_back();
            }
        }
        return this->head.back();
    }
    
    inline bool empty() const {
        return this->head.empty() && this->tail.empty();
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: sol/232.py
class MyQueue:

    def __init__(self):
        self.tail: list[int] = list()
        self.head: list[int] = list()

    def push(self, x: int) -> None:
        self.tail.append(x)

    def pop(self) -> int:
        self.peek()
        return self.head.pop()

    def peek(self) -> int:
        if len(self.head) == 0:
            while len(self.tail) != 0:
                self.head.append(self.tail.pop())
        return self.head[-1]

    def empty(self) -> bool:
        return len(self.head) == 0 and len(self.tail) == 0


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()
